src/decoding-common.jl

```diff
@@ -137,56 +137,56 @@ function decode_next(start_idx, bytes::Array{UInt8, 1}, with_iana::Bool)
                     bytes_consumed += sub_bytes_consumed
 
                     big_int = parse(BigInt, bytes2hex(big_int_bytes),
-                                    HEX_BASE)
+                                    base=HEX_BASE)
                     if tag == NEG_BIG_INT_TAG
                         big_int = -(big_int + 1)
                     end
 
-                    const data = big_int
+                    data = big_int
                 else
-                    const data = retrieve_plain_pair()
+                    data = retrieve_plain_pair()
                 end
             else
-                const data = retrieve_plain_pair()
+                data = retrieve_plain_pair()
             end
 
         elseif typ == TYPE_7
-            const addntl_info = first_byte & ADDNTL_INFO_MASK
+            addntl_info = first_byte & ADDNTL_INFO_MASK
             bytes_consumed = 1
 
             if addntl_info < SINGLE_BYTE_SIMPLE_PLUS_ONE + 1
                 bytes_consumed += 1
                 if addntl_info < SINGLE_BYTE_SIMPLE_PLUS_ONE
-                    const simple_val = addntl_info
+                    simple_val = addntl_info
                 else
                     bytes_consumed += 1
-                    const simple_val = bytes[start_idx + 1]
+                    simple_val = bytes[start_idx + 1]
                 end
 
                 if simple_val == SIMPLE_FALSE
-                    const data = false
+                    data = false
                 elseif simple_val == SIMPLE_TRUE
-                    const data = true
+                    data = true
                 elseif simple_val == SIMPLE_NULL
-                    const data = Null()
+                    data = Null()
                 elseif simple_val == SIMPLE_UNDEF
-                    const data = Undefined()
+                    data = Undefined()
                 else
-                    const data = Simple(simple_val)
+                    data = Simple(simple_val)
                 end
             else
                 if addntl_info == ADDNTL_INFO_FLOAT64
-                    const float_byte_len = SIZE_OF_FLOAT64
+                    float_byte_len = SIZE_OF_FLOAT64
                 elseif addntl_info == ADDNTL_INFO_FLOAT32
-                    const float_byte_len = SIZE_OF_FLOAT32
+                    float_byte_len = SIZE_OF_FLOAT32
                 elseif addntl_info == ADDNTL_INFO_FLOAT16
                     error("Decoding 16-bit floats isn't supported.")
                 end
 
                 bytes_consumed += float_byte_len
-                const data = hex2num(bytes2hex(
+                data = reinterpret(Float64,parse(UInt64,bytes2hex(
```

Comment:
You shouldn't need to use `bytes2hex` or `parse` at all here.  For example, you could just do:
```jl
reinterpret(Float64, ntoh(reinterpret(UInt64, bytes[(start_idx + 1):(start_idx + float_byte_len)])))
```
This could be made even more efficient (if you care) by eliminating the creation and reinterpretation of a array for the `bytes` slice:
```jl
reinterpret(Float64, UInt64(bytes[start_idx + 1]) << 56 +
                     UInt64(bytes[start_idx + 2]) << 48 +
                     UInt64(bytes[start_idx + 3]) << 40 +
                     UInt64(bytes[start_idx + 4]) << 32 +
                     UInt64(bytes[start_idx + 5]) << 24 +
                     UInt64(bytes[start_idx + 6]) << 16 +
                     UInt64(bytes[start_idx + 7]) << 8 +
                     UInt64(bytes[start_idx + 8]))
```

Reply:
But if instead of working with `bytes` arrays, you used an `IOBuffer`, you could just do:
```jl
ntoh(read(buffer, Float64))
```
(where the `ntoh` is to fixe the byte order).